### ISSUE TITLE
scipy-notebook Image Missing ipympl?

### ISSUE THREAD

**Issue by j123github** [jupyter/docker-stacks#955]
It appears that the scipy-notebook image is missing ipympl, resulting in `%matplotlib widget` failing to run. As a result, interactive plots are not easily available when running Jupyter Lab.

It wasn't clear to me that this was a deliberate omission, since the jupyter-widgets extension is being installed [here](https://github.com/jupyter/docker-stacks/blob/8bfffcaa3f41d8493a71648399bb540e8aa24860/scipy-notebook/Dockerfile#L51). Apologies in advance if this is an explicit choice and I have misunderstood.

I have added the installation of `ipympl=0.3.3` to a local scipy-notebook Dockerfile and have been using `%matplotlib widget` without errors.

**What docker image you are using?**

`jupyter/scipy-notebook` or `jupyter/datascience-notebook`

**What complete docker command do you run to launch the container (omitting sensitive values)?**

`docker run -p 8888:8888 jupyter/scipy-notebook:latest start.sh jupyter lab`

or

`docker run -p 8888:8888 jupyter/datascience-notebook:latest start.sh jupyter lab`

**What steps do you take once the container is running to reproduce the issue?**

Example:

1. Visit http://localhost:8888/lab
2. Start a Python notebook in Jupyter Lab
3. Run 
```
%matplotlib widget
import matplotlib.pyplot as plt
```

**What do you expect to happen?**

For the command to complete without error.

**What actually happens?**

The following error is output due to ipympl missing:
```
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-1-eeaad41e1227> in <module>
----> 1 get_ipython().run_line_magic('matplotlib', 'widget')
      2 import matplotlib.pyplot as plt

/opt/conda/lib/python3.7/site-packages/IPython/core/interactiveshell.py in run_line_magic(self, magic_name, line, _stack_depth)
   2312                 kwargs['local_ns'] = sys._getframe(stack_depth).f_locals
   2313             with self.builtin_trap:
-> 2314                 result = fn(*args, **kwargs)
   2315             return result
   2316 

</opt/conda/lib/python3.7/site-packages/decorator.py:decorator-gen-108> in matplotlib(self, line)

/opt/conda/lib/python3.7/site-packages/IPython/core/magic.py in <lambda>(f, *a, **k)
    185     # but it's overkill for just that one bit of state.
    186     def magic_deco(arg):
--> 187         call = lambda f, *a, **k: f(*a, **k)
    188 
    189         if callable(arg):

/opt/conda/lib/python3.7/site-packages/IPython/core/magics/pylab.py in matplotlib(self, line)
     97             print("Available matplotlib backends: %s" % backends_list)
     98         else:
---> 99             gui, backend = self.shell.enable_matplotlib(args.gui.lower() if isinstance(args.gui, str) else args.gui)
    100             self._show_matplotlib_backend(args.gui, backend)
    101 

/opt/conda/lib/python3.7/site-packages/IPython/core/interactiveshell.py in enable_matplotlib(self, gui)
   3412                 gui, backend = pt.find_gui_and_backend(self.pylab_gui_select)
   3413 
-> 3414         pt.activate_matplotlib(backend)
   3415         pt.configure_inline_support(self, backend)
   3416 

/opt/conda/lib/python3.7/site-packages/IPython/core/pylabtools.py in activate_matplotlib(backend)
    311     matplotlib.rcParams['backend'] = backend
    312 
--> 313     import matplotlib.pyplot
    314     matplotlib.pyplot.switch_backend(backend)
    315 

/opt/conda/lib/python3.7/site-packages/matplotlib/pyplot.py in <module>
   2353     dict.__setitem__(rcParams, "backend", rcsetup._auto_backend_sentinel)
   2354 # Set up the backend.
-> 2355 switch_backend(rcParams["backend"])
   2356 
   2357 # Just to be safe.  Interactive mode can be turned on without

/opt/conda/lib/python3.7/site-packages/matplotlib/pyplot.py in switch_backend(newbackend)
    219         else "matplotlib.backends.backend_{}".format(newbackend.lower()))
    220 
--> 221     backend_mod = importlib.import_module(backend_name)
    222     Backend = type(
    223         "Backend", (matplotlib.backends._Backend,), vars(backend_mod))

/opt/conda/lib/python3.7/importlib/__init__.py in import_module(name, package)
    125                 break
    126             level += 1
--> 127     return _bootstrap._gcd_import(name[level:], package, level)
    128 
    129 

ModuleNotFoundError: No module named 'ipympl'
```

**Comment by parente** [comment]
The image definitions here pre-date `ipympl`. Since we already install JupyterLab, IPython widgets, and Matplotlib, it makes sense to me that `ipympl` should also be included. Feel free to send a PR.

**Comment by mpmdean** [comment]
I strongly agree and would this to start this process

**Comment by romainx** [comment]
Hello  @j123github 
 
@mpmdean has worked on this feature and we are happy to say that it has been merged (#1039) and it's working now. You can pull the latest versions and try it.

Could you confirm that it's OK for you?

**Comment by romainx** [comment]
I'm closing it since it's done. Feel free to open a new issue in case of problem.
